comp/core/workloadmeta/collectors/internal/process/process_collector_test.go

```diff
@@ -0,0 +1,341 @@
+		CreationTime: time.UnixMilli(proc.Stats.CreateTime).UTC(),
+	}
+	return proc, expectedProc
+}
+
+func createTestProcess2(pid int32, createTime int64) (*procutil.Process, *workloadmeta.Process) {
+	proc := &procutil.Process{
+		Pid:     pid,
+		Ppid:    9,
+		NsPid:   3,
+		Name:    "some name 2",
+		Cwd:     "some_directory/path/path2",
+		Exe:     "exe",
+		Comm:    "hello",
+		Cmdline: []string{"c1", "c2", "c3", "c4", "c5"},
+		Uids:    []int32{1},
+		Gids:    []int32{1, 2},
+		Stats: &procutil.Stats{
+			CreateTime:  createTime,
+			Status:      "",
+			Nice:        0,
+			OpenFdCount: 0,
+			NumThreads:  0,
+			CPUPercent:  nil,
+			CPUTime:     nil,
+			MemInfo:     nil,
+			MemInfoEx:   nil,
+			IOStat:      nil,
+			IORateStat:  nil,
+			CtxSwitches: nil,
+		},
+	}
+
+	expectedProc := &workloadmeta.Process{
+		EntityID: workloadmeta.EntityID{
+			Kind: workloadmeta.KindProcess,
+			ID:   strconv.Itoa(int(proc.Pid)),
+		},
+		Pid:          proc.Pid,
+		Ppid:         proc.Ppid,
+		NsPid:        proc.NsPid,
+		Name:         proc.Name,
+		Cwd:          proc.Cwd,
+		Exe:          proc.Exe,
+		Comm:         proc.Comm,
+		Cmdline:      proc.Cmdline,
+		Uids:         proc.Uids,
+		Gids:         proc.Gids,
+		CreationTime: time.UnixMilli(proc.Stats.CreateTime).UTC(),
+	}
+	return proc, expectedProc
+}
+
+// TestCreatedProcessesCollection tests the collector capturing new processes
+func TestCreatedProcessesCollection(t *testing.T) {
+	collectionInterval := time.Second * 10
+
+	creationTime1 := time.Now().Unix()
+	pid1 := int32(1234)
+	proc1, expectedProc1 := createTestProcess1(pid1, creationTime1)
+
+	creationTime2 := time.Now().Add(time.Second).Unix()
+	pid2 := int32(9999)
+	proc2, expectedProc2 := createTestProcess2(pid2, creationTime2)
+
+	for _, tc := range []struct {
+		description        string
+		configOverrides    map[string]interface{}
+		processesToCollect map[int32]*procutil.Process
+		expectedProcesses  map[int32]*workloadmeta.Process
+	}{
+		{
+			description:     "single new process",
+			configOverrides: map[string]interface{}{},
+			processesToCollect: map[int32]*procutil.Process{
+				proc1.Pid: proc1,
+			},
+			expectedProcesses: map[int32]*workloadmeta.Process{
+				expectedProc1.Pid: expectedProc1,
+			},
+		},
+		{
+			description:     "multiple new processes",
+			configOverrides: map[string]interface{}{},
+			processesToCollect: map[int32]*procutil.Process{
+				proc1.Pid: proc1,
+				proc2.Pid: proc2,
+			},
+			expectedProcesses: map[int32]*workloadmeta.Process{
+				expectedProc1.Pid: expectedProc1,
+				expectedProc2.Pid: expectedProc2,
+			},
+		},
+	} {
+		t.Run(tc.description, func(t *testing.T) {
+			c := setUpCollectorTest(t, tc.configOverrides)
+			ctx, cancel := context.WithCancel(context.TODO())
+			defer cancel()
+
+			// TODO: we should use Start() instead of 3 lines below when configuration is sorted as Start() is currently
+			// by default disabled
+			c.collector.store = c.mockStore
```

Comment:
Nit, optional: you already set `collector.store` in the constructor on line 52. I would recommend actually removing `store` from the constructor parameters, as when we are using `Start()` everywhere we won't ever be passing it into the constructor.